### PR TITLE
chore: Add unauthenticated API integration test

### PR DESCRIPTION
**Pull request by sichanyoo**
## Issue \#

- https://github.com/awslabs/aws-sdk-swift/issues/1677

## Description of changes

- Adds an integration test for an API call that doesn't use authentication

## Conventional Commits

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: IntegrationTests/Services/AWSCognitoIdentityIntegrationTests/UnauthenticatedAPITests.swift [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        _ = try await cognitoIdentityClient.deleteIdentityPool(input: DeleteIdentityPoolInput(
+            identityPoolId: identityPoolID
+        ))
+    }
+
+    func testUnauthenticatedAPI() async throws {
+        _ = try await cognitoIdentityClient.getId(
+            input: GetIdInput(accountId: accountID, identityPoolId: identityPoolID)
+        )
+        // API call completed successfully
```
Comment (dayaffe) [inline IntegrationTests/Services/AWSCognitoIdentityIntegrationTests/UnauthenticatedAPITests.swift:49]: should we check that the deserialized response contains expected results?

Comment (sichanyoo) [reply IntegrationTests/Services/AWSCognitoIdentityIntegrationTests/UnauthenticatedAPITests.swift:49]: sure, though reaching line 49 would mean the operation succeeded without throwing error, it wouldn't hurt to check it. Adopting suggestion.

**Review by dayaffe — APPROVED** [review]

**Review by jbelkins — COMMENTED** [review]
The point of this is to prove that unauthenticated requests can be performed successfully.  But the machine performing integration tests will have a set of AWS credentials configured for use in all the other tests.

How do we show that this request was generated unauthenticated, and isn't just succeeding because there happen to be credentials on the machine?

**Comment by sichanyoo** [comment]
@jbelkins 

We know for sure that the operation used in this new integration test doesn't use authentication and is sending an unauthenticated request because the generated auth scheme resolver for `CognitoIdentity`'s `getId` operation returns only a single auth scheme `smithy.api#noAuth`. (see [here](https://github.com/awslabs/aws-sdk-swift/blob/c223c13e0f49693398733ae0689cbaf3e870295b/Sources/Services/AWSCognitoIdentity/Sources/AWSCognitoIdentity/AuthSchemeResolver.swift#L39-L40)). 

This means when the auth scheme gets resolved during runtime, it resolves as a "no-auth" auth scheme in `AuthSchemeMiddleware` [here](https://github.com/smithy-lang/smithy-swift/blob/5f6949b2677130730ed0062835b6e9e57e8a288c/Sources/ClientRuntime/Networking/Http/Middlewares/AuthSchemeMiddleware.swift#L55-L62), which then subsequently results in no signing occurring during `SignerMiddleware` [here](https://github.com/smithy-lang/smithy-swift/blob/5f6949b2677130730ed0062835b6e9e57e8a288c/Sources/ClientRuntime/Networking/Http/Middlewares/SignerMiddleware.swift#L31-L34).

**Review by jbelkins — APPROVED** [review]

**Comment by jbelkins** [comment]
A check was added to the test to ensure that the HTTP `Authentication:` header is not set when making an unauthenticated request.  In response, I have approved.

**Pull request merged**